tests/components/fritzbox/test_init.py

```diff
@@ -0,0 +1,83 @@
+"""Tests for the AVM Fritz!Box integration."""
+from unittest.mock import Mock, call
+
+from homeassistant.components.fritzbox.const import DOMAIN as FB_DOMAIN
+from homeassistant.components.switch import DOMAIN as SWITCH_DOMAIN
+from homeassistant.config_entries import ENTRY_STATE_LOADED, ENTRY_STATE_NOT_LOADED
+from homeassistant.const import CONF_DEVICES, CONF_HOST, CONF_PASSWORD, CONF_USERNAME
+from homeassistant.helpers.typing import HomeAssistantType
+from homeassistant.setup import async_setup_component
+
+from . import MOCK_CONFIG, FritzDeviceSwitchMock
+
+from tests.common import MockConfigEntry
+
+
+async def test_setup(hass: HomeAssistantType, fritz: Mock):
+    """Test setup of integration."""
+    assert await async_setup_component(hass, FB_DOMAIN, MOCK_CONFIG)
+    await hass.async_block_till_done()
+    entries = hass.config_entries.async_entries()
+    assert entries
+    assert entries[0].data[CONF_HOST] == "fake_host"
+    assert entries[0].data[CONF_PASSWORD] == "fake_pass"
+    assert entries[0].data[CONF_USERNAME] == "fake_user"
+    assert fritz.call_count == 1
+    assert fritz.call_args_list == [
+        call(host="fake_host", password="fake_pass", user="fake_user")
+    ]
+
+
+async def test_setup_duplicate_config(hass: HomeAssistantType, fritz: Mock, caplog):
+    """Test duplicate config of integration."""
+    DUPLICATE = {
+        FB_DOMAIN: [
+            MOCK_CONFIG[FB_DOMAIN][CONF_DEVICES][0],
```

Comment:
We're missing the `CONF_DEVICES` key in the domain dict that we create.

Reply:
To be honest I don't understand what you mean. I reverted the yaml config changes and instead deprecated the yaml. This makes for me much more sense than changing it and deprecate it in the next release. So the CONF_DEVICES is back in the yaml.

Reply:
Right, it's back so we need to include it here in the dictionary that we create.

```py
DUPLICATE = {
    FB_DOMAIN: {
        CONF_DEVICES: [
            MOCK_CONFIG[FB_DOMAIN][CONF_DEVICES][0],
            ...
        ],
    },
}
```

Reply:
Ahhh yes, forgotten. This test is failing anyway with the deprecation.